Pixel Movement Support: Tiles? What are those?

Couls opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds basic pixel movement support for most of the systems in the game
Some things still use tile based movement:
AI camera and things that use that kind of systems, for example the xenobio camera console, holocalls
Phazon phasing, as this would require a lot more work to move over to pixel movement due to how CanPass works

Things to keep in mind:
Players will have a slightly longer reach due to not being centered to tiles anymore
You can now bump into multiple objects(walking into the center of airlocks will open both airlocks you come in contact with)
Lying down does not change your hitbox(yet but support could be added to accomodate for this later)
Movement will be slower as you no longer move entire tiles at a time, take some time to adjust to this change!
Playing at 60 fps should be smooth now and not jittery!
Mob hitboxes aren't their whole sprite, its mainly their lower half(use your feet as measurement) this is to make it easier to go through airlocks and avoid slipping over things that wouldn't be underneath you/touching your legs

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage 
maintainers from merging your PR! -->

## Why It's Good For The Game
Things that it can bring to the game: Being able to walk around soap and other traps on the floor without having to move them from the tile(squeezing around them or hugging the wall). Being able to fit more crates on the cargo shuttle, Being able to squeeze between closets etc.

Pixel movement feels like the future of ss13 and its movement system in general, because not only would it allow for the things listed above but there is a lot of potential to expand things like the combat system to swing in a pixel range and allow people to dodge swings by moving away at just the right time
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: Coul, ninjanomnom, silicons, tralezab
add: Pixel movement! A massive increase in the bluespace density around the station has seemed to cause a change to the way things work on the station..
add: Move carefully action, hold shift to slow to a crawl and get pixel precision movement
del: Removed the windows on the conveyor belt in metastation cargo as it was blocking crates from passing through them in pixel movement
add: New edge sliding preference, makes it so you automatically slide along corners you walk into if more than half of your hitbox is past the edge
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Couls committed:
Pixel Movement

ShibaInuLord commented:
this is so fucking cursed in testing holy shit

Couls commented:
![image](https://user-images.githubusercontent.com/30060146/88446171-bffe0a00-cdf5-11ea-9879-13a28f14f48f.png)
![image](https://user-images.githubusercontent.com/30060146/88446175-c8564500-cdf5-11ea-9b76-c07511a0ae96.png)
![image](https://user-images.githubusercontent.com/30060146/88446183-cf7d5300-cdf5-11ea-924a-9888be02e0e0.png)

EdgeLordExe commented:
I LOVE THIS

Mycah142 commented:
This is a joke right? A jape? This is just trollin'. You can't be serious.

This'll upset literally everyone. I informally request a speedmerge.

Jared-Fogle commented:
Pulling is very annoying, especially with things like lockers that pull slower than you, they also regularly get caught on dense objects.

Fireman carrying makes the person being carried look very far away if you kick from where you are and is very jittery.

![image](https://user-images.githubusercontent.com/35135081/88446507-3e9b7d00-cddf-11ea-8de0-03c86f2d3c72.png)

Couls commented:
> Pulling is very annoying, especially with things like lockers that pull slower than you, they also regularly get caught on dense objects.
Lockers and other things should try to sidestep any dense objects that aren't mobs or anchored
> Fireman carrying makes the person being carried look very far away if you kick from where you are and is very jittery.
Can't reproduce the kicking thing can you provide more information?
![image](https://user-images.githubusercontent.com/30060146/88446568-17ea4000-cdf9-11ea-85ae-bcec02b88656.png)

Jared-Fogle commented:
> Can't reproduce the kicking thing can you provide more information?
My reproduction was simply disarming them to drop them on the ground, starting the carry, and continuing to disarm them while they were on the ground until it carried them. Then, after walking around, their position was desynced.

Couls commented:
> My reproduction was simply disarming them to drop them on the ground, starting the carry, and continuing to disarm them while they were on the ground until it carried them. Then, after walking around, their position was desynced.

Are you sure the carry didn't fail? they aren't really ontop of you and I couldn't manage to carry someone while keeping them down with shoves in my test it kept failing

Jared-Fogle commented:
I tried it several times and I'm pretty sure it worked. I could be mistaken though.

Couls commented:
I'll see what I can do to make shoving a bit smoother

LemonInTheDark commented:
Holy god damn hell they said it couldn't be done. You fucking madlad

IndieanaJones commented:
This unnerves me.  Nice work.  Though, I have to ask, how does this work with things such as bodyblocking people, megafauna fights, etc...  Is there any things we can't do anymore with this sort of system?

Couls commented:
> This unnerves me. Nice work. Though, I have to ask, how does this work with things such as bodyblocking people, megafauna fights, etc... Is there any things we can't do anymore with this sort of system?

The idea is that you can't really be bodyblocked anymore because you can usually just go around most obstacles, since theres a much larger range of moving around areas. The main downside to this system to playing is probably that the movespeed is going to be significantly slower and there's not really much you can do about that as moving entire tiles looks very janky(think when you don't remove step_ values from mapedits before pixel movement) another possible downside is single step proc calls are VERY small and that used to be our main method of movement. Now we have to using walk procs and those don't produce results immediately as it processes in the background

ATH1909 commented:
cursed

... but intriguing.

Couls committed:
Banana bounds, keybind signals for move carefully, make buckling easier,

nearest turf proc is now global and finds the nearest turf to the AM passed as an argument

ATH1909 commented:
> add: Move carefully action, hold shift to slow to a crawl and get pixel precision movement

Can we bind this to something other than shift? Slowing to a crawl whenever I try to open a door as a silicon or use a quickdraw hotkey is going to be quite annoying (and will probably result in me immediately rebinding that slow to a crawl action to something other than shift).

How does this interact with chameleon projectors?

Do speed boosts look okay? If you stack enough speed modifiers, can you phase through something thin, like a window?

Couls commented:
> Can we bind this to something other than shift? Slowing to a crawl whenever I try to open a door as a silicon or use a quickdraw hotkey is going to be quite annoying (and will probably result in me immediately rebinding that slow to a crawl action to something other than shift).

Yes it's a keybind

> How does this interact with chameleon projectors?

Chameleon projectors should work similar to how they did prior to pixel move in regards to slowdown as the temperature gets lower while still never being faster than someone who isn't chameleon as of next commit

> Do speed boosts look okay? If you stack enough speed modifiers, can you phase through something thin, like a window?

You can't phase through anything and you move more pixels with speed boosts and move less pixels with slowdowns

Couls committed:
chameleonproj pixel movement support

good find ATH

Couls committed:
trailing newline

Geikkamir commented:
Honestly I really like the visual clarity that tiles provide. Even with a lot of the jank-ass sprites in this game tiles make it easy to tell at a glance if something is close enough to interact with, or to attack you, or if something is obstructing your path, etc.

Couls commented:
> Honestly I really like the visual clarity that tiles provide. Even with a lot of the jank-ass sprites in this game tiles make it easy to tell at a glance if something is close enough to interact with, or to attack you, or if something is obstructing your path, etc.

This is a very valid point, we could always make use of highlighting interactable objects though, and things obstructing your path would still be clear in this situation

Couls committed:
Shoving moves half a tile, a few misc fixes(basketball net, stacks, adjacency)

MrDoomBringer commented:
Honestly, I'm absolutely against this. A huge part of combat is this game relies on knowing that everything takes up exactly one tile. I can have my mouse aimed at the exact position someone will be when they round a corner, or I can block someone chasing me through maint by dragging a dense object and holding it between me and them. 

In my eyes, this removes a lot of depth without providing that much clear benefit. Still, I guess it's shiny and cool so it'll probably end up getting merged anyways

LemonInTheDark commented:
sounds like  we should change combat to be less pixel hunty

Couls commented:
On the contrary I think this adds a lot more depth, having to walk around people instead of magically swapping places with them, having to actually spend time blocking paths instead of an half-tile closet making it impossible for you to get by. Having to walk around soaps or move out of the way of a throw that just barely near misses you is all experiences that are just made better with pixel movement over tile based movement. It changes things from the way you've always known them but I disagree with statements saying that this will limit people I feel like it'll actually allow people to be more creative with approaching things, like cargo trying to fit the maximum amount of crates in their shuttle

TheVekter commented:
YOU WERE TOO BUSY TRYING TO FIND OUT IF YOU COULD INSTEAD OF ASKING IF YOU SHOULD

SOME THINGS ARE BEST LEFT UNTAMPERED WITH

Partheo commented:
"Your scientists were so preoccupied with whether or not they could, they didn’t stop to think if they should."

Why the hell did you think this was a good idea? I didn't think I'd ever see this in SS13. But SS13 has been built from the ground up as a tile-based game. Changing this now and allowing more freedom of movement is not going to improve the game. I think this PR is a great example of adding something to the game that is entirely unnecessary and actively goes against any groundwork that currently exists. I'm impressed that you manage to push the technology this far, but this is adding stuff for the sake of adding with no regard to how the game is designed.

A tile based game is tile-based for a reason. Adding free movement to a game that has been designed as tile-based game doesn't make any sense. 

If a dev working on a turn-based game suddenly made a suggestion to make it real-time instead of turn-based halfway through development, he'd be laughed at by everyone. "It's more dynamic!" "It gives you more freedom!" Yes, it does. No one would deny that. But the game is a turn-based game, you can't just make it real-time and say that the game is better now.

Geikkamir commented:
>  having to walk around people instead of magically swapping places with them

Is this really good gameplay? It's more annoying, sure, but in what way is that a positive? It seems like just a straight QoL downgrade.

> having to actually spend time blocking paths instead of an half-tile closet making it impossible for you to get by

You can already get past by just opening and/or grabbing the locker

> Having to walk around soaps or move out of the way of a throw that just barely near misses you is all experiences that are just made better with pixel movement over tile based movement

This is kind of an example of what I was talking about in terms of visual clarity. With a tile-based system, I can very easily tell at a glance if a banana peel or soap bar on a tile is going to slip me, and when throwing down a slip I can easily tell what area I'm denying by doing so. With a pixel-based system, there might be times I think that I'm clear but actually the bottom pixel of my foot is going to collide with the top pixel of the banana peel in front of me. I also have to worry about the hitbox of both my own sprite (which if I understand correctly you're saying does not line up with the entire sprite itself) and the hitbox of other objects. This also raises the concern of whether this system will turn the size of sprites of some objects into a significant balance issue.

> like cargo trying to fit the maximum amount of crates in their shuttle

In all my time playing cargo, I don't think I have literally ever had a problem with fitting crates into the shuttle. You are given more than enough space to fit as much stuff as you would ever reasonably need to in one trip.

PKPenguin321 commented:
I do agree that this will be weird with our still-tile-based melee combat, but that shit sucks regardless and should be replaced in the near future with something better anyways

Couls commented:
> Why the hell did you think this was a good idea? I didn't think I'd ever see this in SS13. But SS13 has been built from the ground up as a tile-based game. Changing this now and allowing more freedom of movement is not going to improve the game. I think this PR is a great example of adding something to the game that is entirely unnecessary and actively goes against any groundwork that currently exists. I'm impressed that you manage to push the technology this far, but this is adding stuff for the sake of adding with no regard to how the game is designed.

I paid a lot of attention to the original design of many things and tried to keep them working as similar as they did with tile based movement while still making improvements where there could be some without being too intrusive, almost all features added are entirely unnecessary and the game could work perfectly as is now and this doesn't actually go against the groundwork but create more groundwork and improve on it, everything that was tile based can be made better using pixel movement and if not could actually still stay tile based, like the AI and xenobio console. These things still move tile by tile

> A tile based game is tile-based for a reason. Adding free movement to a game that has been designed as tile-based game doesn't make any sense.

To pave the path for improvements on the game that wouldn't be as good or even possible with tile movement, things like swing combat would just be a flat buff if we made it swing on every turf in a direction or something like that, making it even harder to dodge attacks to solve pixel hunting

> If a dev working on a turn-based game suddenly made a suggestion to make it real-time instead of turn-based halfway through development, he'd be laughed at by everyone. "It's more dynamic!" "It gives you more freedom!" Yes, it does. No one would deny that. But the game is a turn-based game, you can't just make it real-time and say that the game is better now.

It's not made better just BECAUSE of pixel movement but it's going to be made better by what pixel movement allows which is a lot of stuff that can only been seen and experienced. It's different and it's strange and I know that but until you get your hands on it you won't really know how much better it feels or smoother or natural because these are these that can't be described or explained in screenshots or videos things like responsiveness and smoothness have to be seen first hand

Geikkamir commented:
> To pave the path for improvements on the game that wouldn't be as good or even possible with tile movement, things like swing combat would just be a flat buff if we made it swing on every turf in a direction or something like that, making it even harder to dodge attacks

How is it any less of a buff in a pixel movement system? In either case it's making it harder to dodge attacks because you don't have to click on the janky, rapidly moving sprite of your opponent, which is the whole point.

> It's not made better just BECAUSE of pixel movement but it's going to be made better by what pixel movement allows which is a lot of stuff that can only been seen and experienced. It's different and it's strange and I know that but until you get your hands on it you won't really know how much better it feels or smoother or natural because these are these that can't be described or explained in screenshots or videos things like responsiveness and smoothness have to be seen first hand

I actually have tried a system like this on Corporate Mercenary. It's kinda terrible. Like one would expect, it adds virtually nothing to the game that couldn't be done with tiles and it also adds a lot of little ambiguities that make it harder to parse at a glance what is in range of what, what is going to collide with what, etc.

Couls commented:
> How is it any less of a buff in a pixel movement system? In either case it's making it harder to dodge attacks because you don't have to click on the janky, rapidly moving sprite of your opponent, which is the whole point.

If we had swing combat you'd only have to click in a direction instead of hunting the sprite of your opponent.

> I actually have tried a system like this on Corporate Mercenary. It's kinda terrible. Like one would expect, it adds virtually nothing to the game that couldn't be done with tiles and it also adds a lot of little ambiguities that make it harder to parse at a glance what is in range of what, what is going to collide with what, etc.

It seems like you just need to get a better feel of the size of things, everything should kind of fit in its sprite so if it looks like its going to collide with something it likely will I can't really say i've had trouble with this in my tests and this isn't really an objective thing I think you just need more time to adjust to the difference, i'm not really sure what you mean by "adding things that couldn't be done by tile based movement" because this PR is mainly support for the system and adding a bunch of pixel movement exclusive bloat would make this impossible to review, the whole purpose of this PR is to support pixel movement and improve things that use it or didn't prior, rather than adding content that's pixel movement exclusive immediately. I'm not out to add a swing combat system or a crazy vector movement inertia system or ships that pixel move through space this is all way of out scope of this specific PR but would probably be possible with enough work after and if this was merged. What i'm providing is pixel movement that works with the content we have now and is much smoother looking and feeling. There's a lot more side effects that come from pixel movement but I truly believe the majority of not all of it is beneficial to the game

MrDoomBringer commented:
> having to actually spend time blocking paths instead 

You're wording this like it's a great balance change when in reality it's going to take more time because players will have to spend a while fussing around with pixel-movement and an overcomplicated hitbox system.

>Having to walk around soaps or move out of the way of a throw that just barely near misses you

But we can do all these things already? Also, one tile wide hallways have always been an intentionally riskier place to travel because you're nearly guaranteed to get hit if someone's shooting after you, something important to consider when picking a path to take when escaping from someone. This feature will (to a degree, granted) take away that depth.

> like cargo trying to fit the maximum amount of crates in their shuttle

Unsurprisingly enough, loading the cargo shuttle is going to be made much more annoying and unintuitive with this merge. Instead of being easily able to tell how many more crates can fit in the shuttle, players are going to have to finagle with fitting the crates in while worrying about "oops this space is 1 pixel too tight to fit a crate in sorry".

Geikkamir commented:
> If we had swing combat you'd only have to click in a direction instead of hunting the sprite of your opponent.

I know, I'm saying I don't see how pixel movement is a requirement for swing combat or how it's any less of a buff under a pixel movement system. Goon essentially allows for the same thing with special attacks and it's tile-based.

> It seems like you just need to get a better feel of the size of things, everything should kind of fit in its sprite so if it looks like its going to collide with something it likely will I can't really say i've had trouble with this in my tests

This isn't too difficult when things are calm and time isn't an issue, but where it really becomes a problem is when things are moving fast and there's a lot going on on the screen at once.

Also like I mentioned, there's the problem that this makes the sprite size of any object that cares about collision a gameplay concern, which is especially tricky considering how wildly inconsistent our sprites are. Consider, for example, how enormous the sprite for a bar of soap is compared to a banana peel.

Couls commented:
> > having to actually spend time blocking paths instead
> 
> You're wording this like it's a great balance change when in reality it's going to take more time because players will have to spend a while fussing around with pixel-movement and an overcomplicated hitbox system.

You won't know this until you see it in action


> But we can do all these things already? Also, one tile wide hallways have always been an intentionally riskier place to travel because you're nearly guaranteed to get hit if someone's shooting after you, something important to consider when picking a path to take when escaping from someone. This feature will (to a degree, granted) take away that depth.

My whole point isn't that you couldn't do this before my point is that this improves the experience and feels more like you've dodged something than that you moved to the correct tile in time, shots will almost definitely still hit you in a one tile wide hallway, again something you'd have only known if you've tried this out

> Unsurprisingly enough, loading the cargo shuttle is going to be made much more annoying and unintuitive with this merge. Instead of being easily able to tell where to drag a crate in the shuttle, players are instead going to have to finagle with fitting the crates in while worrying about "oops this space is 1 pixel too tight to fit a crate in sorry". Of course, this applies to anytime you want to drag something anywhere.

Dragging actually automatically handles small pixel blockages when dragging objects so it won't actually be cumbersome to drag things around. Things not fitting in places where they're too big though is a very realistic issue and a problem that may be fun to solve as a cargo tech, but if you don't like it like someone said earlier the shuttle already has more than enough space and it's literally not any more cumbersome to load it than it was on tile movement, for most situations pixel movement is an improvement to the way things used to work without changing them up too severely which is something I tried to pay attention to a lot in my time working on this, of course this is not 100% true for every scenario but this was a large project and there may have been things I missed.

> Edit:
> 
> > If we had swing combat you'd only have to click in a direction instead of hunting the sprite of your opponent.
> 
> If a reworked combat system is needed with this PR, then do that first. I wouldn't add a PR and say "by the way, this makes the game less fun until I eventually rework this other major system".

My intention isn't to rework the combat system and I don't think this PR makes the game less fun with the current combat system, it's either the same or better(in my opinion) and if you find a specific scenario where it isn't then point it out to me and we could talk about this and possibly work together to improve that situation or behaviour

Couls commented:
> I know, I'm saying I don't see how pixel movement is a requirement for swing combat or how it's any less of a buff under a pixel movement system. Goon essentially allows for the same thing with special attacks and it's tile-based.

Goons solution still has you hunting for the correct tile to click, I think we could make an even better implementation that attacks in an angle from where you click and hits everybody in the swing area and knocks them back a bit this completely eliminates pixel hunting but this is getting off-topic and is out of the scope of this PR, I'm not going to try to use future possibilites as a selling point for this PR despite that being the strongest pushing force and reason I actually wanted to work on this myself.

> This isn't too difficult when things are calm and time isn't an issue, but where it really becomes a problem is when things are moving fast and there's a lot going on on the screen at once.

Pixel hunting is also difficult to do when lots is going on, the panic during the scramble of the battlefield is part of the fun imo. But if you think theres stuff that makes this significantly worse specifically on pixel movement then please let me know so I could look into the issue and ways to improve it

> Also like I mentioned, there's the problem that this makes the sprite size of any object that cares about collision a gameplay concern, which is especially tricky considering how wildly inconsistent our sprites are. Consider, for example, how enormous the sprite for a bar of soap is compared to a banana peel.

Sprites could later be downsized or changed to be standardized so all slipping items are within a rough size area this actual makes sense to me and I don't think this specific issue is enough reason to not swap over to pixel movement while it might be initially annoying on first implementation, we can also scale down sprites using matrixes

bobbahbrown commented:
how do brooms work in this system

Partheo commented:
> It's not made better just BECAUSE of pixel movement but it's going to be made better by what pixel movement allows which is a lot of stuff that can only been seen and experienced. It's different and it's strange and I know that but until you get your hands on it you won't really know how much better it feels or smoother or natural because these are these that can't be described or explained in screenshots or videos things like responsiveness and smoothness have to be seen first hand

I'm sorry, but I'm just not convinced. I know this opens a million doors for SS13, but they are doors that don't need opening. Skyrim wouldn't be a better game if you had a jetpack and could fly anywhere you wanted in the map. More freedom is not necessarily good.

It's like this issue, which will be familiar if you listen to gamedev news:
### not everything needs to be an open-world game

MrDoomBringer commented:
>You won't know this until you see it in action

haha, what? I literally checked out your branch and "saw this all in action" myself before I made my first comment, why would you assume I hadn't?

>shots will almost definitely still hit you in a one tile wide hallway, again something you'd have only known if you've tried this out

Again, I did try this. And as I said earlier, it really is much easier to dodge shots in a 1-wide than it was before (as it was impossible, as it should be, before)

>Dragging actually automatically handles small pixel blockages 

Not when you're pushing one object into place between others, which is the situation I was describing (read: the cargo shuttle)

Couls commented:
> haha, what? I literally checked out your branch and "saw this all in action" myself before I made my first comment, why would you assume I hadn't? My point stands, you have to fuss around with trying to move objects into place now - something that seems minor on paper but in-game can really take away from the experience.

if you want to place an object you're dragging on a specific position on the tile simply click on the tile and it'll work like it did prior to tile movement this takes seconds and would be just as fast as it would be on tile based movement and you shouldn't lose any object you're pulling even through 1 tile small passageways and if there's a scenario where this can happen please report it so I can fix it on the PR and it sounds like you haven't tried it all yourself because a lot of the complaints you stated seem kind of ballpark, if you have an issue with pulling in specific then voicing it can help me fix it and make this transition fit better for you and whatever concerns you may have with it

> Again, I did try this. And as I said earlier, it really is much easier to dodge shots in a 1-wide than it was before (as it was impossible, as it should be, before)

This can easily be fixed but I actually prefer it this way, regardless this isn't a problem of moving to pixel movement as you can still  be hit in a 1 tile wide passageway even with pixel movement

> Not when you're pushing one object into place between others, which is the situation I was describing (read: the cargo shuttle)

I explained that the cargo shuttle already has more than enough space that you don't need to squeeze it to the brim unless you want to for fun and a crate not fitting where it doesn't have space still makes sense to me and doesn't actually sound like a problem

SweptWasTaken commented:
This is obviously a massive change which impacts almost every job and antag class in the game. If the Headcoders and Headmins are going to take this pr seriously, could we have a test server for pixel movement? ~~And maybe a controversial pr test server in general.~~

Just so people could hop on and test it out instead of forming an opinion with only a glance?

SS14 has had a ton of issues "selling" pixel-movement, but most of it's criticism is coming from people who've only looked at it and formed an opinion without testing it out.

Couls commented:
Based swept, someone recommended loading this up on campbell and I totally agree with this

MrDoomBringer commented:
![image](https://user-images.githubusercontent.com/29008542/88468566-ad4a0a80-ceb3-11ea-9f9b-59e76ca16749.png)
>the self upthumb

😔 
<hr/>
Anyways, while I really do respect the (massive) work put into this Pr, I really can't get over the way movement is handled with other objects, ie dragging. It really is just kind of _janky_. Here are some videos I took real quick for people who want to see for themselves real quick:

>if you want to place an object you're dragging on a specific position on the tile simply click on the tile and it'll work like it did prior to tile movement this takes seconds and would be just as fast as it would be on tile based movement

Not gonna lie, from my quick test, click-moving seemed to work well some times and not at all others. [Quick example trying to move around a welder tank](https://streamable.com/w8afus)
<hr/>
>This can easily be fixed but I actually prefer it this way,

How would you fix it? And why should one-tile-wide halls be less dangerous? ([quick video showing what i was talking about here](https://streamable.com/rzspm1))
<hr/>
>I explained that the cargo shuttle already has more than enough space that you don't need to squeeze it to the brim 

Homie this isn't about cargo. It's about the concept of being able to push or drag an object between two other objects. Currently it's very easy to tell when an object can pass between two others, with this PR it isn't. Take these videos for an example:
[Trying to push a closet past a different closet](https://streamable.com/nucqkj)
[Trying to push a closet through an open airlock](https://streamable.com/4cd6vt)

<hr/>
>how do brooms work in this system

[they work perfectly for the most part! A little weird that I could just walk by the trash but I think I'm just used to the old system](https://streamable.com/gs0csl)

Sorry for all the videos and walls of text, I don't mean to hulk-out on your PR or anything. I just really feel like it adds a layer of jankiness (for lack of a better word) at its current stage, and might not add that much enjoyment to the game.

Couls commented:
These videos are very helpful and i'll start working on improving the stuff you pointed out

Partheo commented:
Also, this means that the shape/size of sprites now have an effect on gameplay. Which is... a can of worms to say the least.

MrDoomBringer commented:
@Partheo Couls and I were chatting in discord about this for a bit, hitboxes can be manually controlled as needed, which is fine imo

PKPenguin321 commented:
>Also, this means that the shape/size of sprites now have an effect on gameplay. Which is... a can of worms to say the least.

This is already the case

Couls committed:
larger projectiles, smaller closets, smooth disarm and broom improvements

oh and mouse clicking objects you drag should be a bit better

Couls committed:
More work on dragging stuff, dragging things without density should be MUCH better

Couls committed:
update unit test to use pixel movement

IndieanaJones commented:
I would love to try this out on a server just to see what its like.  If deemed truly a crime against God, it would make an amazing April Fools joke.

necromanceranne commented:
Afros eat more bullet than the average spaceman.

KathyRyals commented:
Removes a lot of emergent gameplay and tricks, doesn't provide enough benefits, that's a naw from me. 
Although this is an amazing feat and I cant imagine the amount of coding needed. Good job on this part

Couls commented:
If you could provide examples of emergent gameplay that it entirely removes that’s important enough to be kept in I could look into ways to bring it back

I don’t know if you tested or for how long you did but the classic tricks you’re used to doing in tile based movement are probably things you didn’t immediately know how to do within your first couple of rounds, i didn’t learn about a lot of tricks until they were used on me and it’s likely that whatever new tricks and emergent gameplay arises from pixel movement will be some time after people get to play with it

Couls commented:
Again if you tried this out privately and you have very specific complaints please voice them because 9 times out of 10 it’s something that can be adjusted or changed to fit pixel movement as well, hit boxes dan be changed and they can even go back to an entire tile size though I’d like to avoid doing that as it defeats the point of having things different sizes

KathyRyals commented:
Hmm thanks for the answer, the first thing that came to my mind(and some others as well) is no longer being able to use some bulky items like lockers / crates to block movement in maintenance / tight corridors. In an effort to put my money where my mouth is, I'll try and list what I feel could be lost with this PR, thank you very much!

Couls committed:
connectors and air pump should be more responsive

Acer202 commented:
Just gave this a shot on a private server with a couple friends to test it out. Half of us thought it was neat, the other half thought it was against god and nature.Personally, I love this. There's a few things I want to note, because giving feedback to this is essential to it having a shot, I think. Here's what we found:

1. You can open lockers on the other side of a glass pane.
2. Mobs move really scary fast. Monkeys are terrifying, and ever-so-slightly faster than you. Having them chase you now becomes a horror film like experience. (I love this, but it causes big changes to mining mobs)
3. The ACME extendo-hand has become a formidable thing. Whether this is good or bad is yet to be seen, but you can shove/grab/hug people from long distances. I also love this.
4. Moving an office chair through a single-tile airlock is nigh impossible at the moment
5. Piggybacking causes a jitter on the carrier for the rider, and on the rider for the carrier, but works fine.
6. We had something weird happen where we managed to push someone through a wall by dragging them and bumping into something. I'll have to work on replicating that.
7. I'm not sure if it's just me or what, but dodging bullets is actually harder.
8. Walking past eachother/shoving someone over in a 1-tile hallway is really neat. Instead of a silly shove joust for 3 minutes, you can just skedaddle around their side and shove them anyways. Shoving is actually kinda buffed by this change.

So far, I really like how it feels. It would be a hard sell, I think, for most players. I feel it would definitely need like a test server to gauge who'd like it, and who'd hate it.

Couls commented:
Very useful notes and I'm glad you like it I'll address some of these and look into the ones that need investigation

> Piggybacking causes a jitter on the carrier for the rider, and on the rider for the carrier, but works fine.

smoothing is handled by the client so when we're being moved by something like a carrier it might be slightly jittery, not sure if I can smooth it any further

> Mobs move really scary fast. Monkeys are terrifying, and ever-so-slightly faster than you. Having them chase you now becomes a horror film like experience. (I love this, but it causes big changes to mining mobs)

Speed is still a variable we have control over and mining mobs should still be able to be outran them and we can work towards slowing down another mobs if they feel too fast with a simple change of a var I do kind of think it makes sense that monkeys would eventually catch up with you if you're unlucky enough to piss one off so i'll leave that one in, they're pretty easy to outsmart/disable as is

> Walking past eachother/shoving someone over in a 1-tile hallway is really neat. Instead of a silly shove joust for 3 minutes, you can just skedaddle around their side and shove them anyways. Shoving is actually kinda buffed by this change.

This kind of interaction is the stuff i'm really excited to see happening and I really do like how natural it would be to end up shoving somebody who just won't get out of the way

I'll look into office chairs, opening closets through panes and the weird phasing through wall glitch later today

Couls committed:
fix chairs and opening closets through walls

Rohesie commented on code/__DEFINES/bounds.dm:
```diff
@@ -0,0 +1,16 @@
+/// Gets the appropriate turf to make step_x/y in range 0-31
+/// ex: step_y 32 will move the turf up one and change step_y to 0
+#define NORMALIZE_STEP(newturf, stepx, stepy) \
+	newturf = locate(newturf.x + FLOOR(stepx/world.icon_size, 1), newturf.y + FLOOR(stepy/world.icon_size, 1), newturf.z);\
```
```suggestion
	newturf = locate(newturf.x + round(stepx/world.icon_size), newturf.y + round(stepy/world.icon_size), newturf.z);\
```
Quirk of the language, `round(x)` acts as `FLOOR(x, 1)`
It's a micro-optimization in case this is used a ton. If you do add this, comment it on the doc.

Also, do multi-line comments like this one with the `/**` format for proper documenting

Rohesie commented on code/__DEFINES/maths.dm:
```diff
@@ -81,6 +81,10 @@
 // amount=0.5 returns the mean of a and b.
 #define LERP(a, b, amount) ( amount ? ((a) + ((b) - (a)) * (amount)) : a )
 
+// Moves the current number towards the target number by amount.
+// It won't overshoot
```
`/**` comment pattern?

Couls replied:
Nothing seems to use this anyway, i'll just get rid of it

Rohesie commented on code/__HELPERS/movement.dm:
```diff
@@ -0,0 +1,94 @@
+/// PIXEL define for general use with pixel movement(converting turf x and y values and dist values to pixel dist)
+#define PIXELS 32
```
```suggestion
#define TILE_SIZE 32
```
Or `PIXEL_TILE_SIZE` ?

Rohesie commented on code/__HELPERS/movement.dm:
```diff
@@ -0,0 +1,94 @@
+
+/// Returns the direction from thingA to thingB in degrees
+/// EAST is 0 and goes counter clockwise
+#define get_deg(thingA, thingB) ATAN2(thingB.true_y() - thingA.true_y(), thingB.true_x() - thingA.true_x())
+
+/// Use this instead of get_dir when things can be on the same turf
+#define get_pixeldir(thingA, thingB) (get_dir(thingA, thingB) || angle2dir(get_deg(thingA, thingB)))
```
Write the macros in caps. `GET_ANGLE` and `GET_PIXEL_DIR` for example.

Valtosin commented:
We testing this on prod, here some issues:

1. Bullets can't hit lying humans and also creates gigabytes of runtimes.
2. You can use buttons behind the windows.
3. Layers:
![image](https://user-images.githubusercontent.com/10297716/88614518-1dc27880-d0ba-11ea-8916-5c297d7f7b55.png)
4. You can push someone on the table and if table is standing near the wall, this will produce broken table and human.

the rest has already been listed before

Rohesie commented on code/datums/action.dm:
```diff
@@ -540,7 +540,7 @@
 		return
 	COOLDOWN_START(src, box_cooldown, 10 SECONDS)
-	var/box = new boxtype(owner.drop_location())
+	var/box = new boxtype(owner.drop_location()[1])
```
type cast `drop_location()`

Rohesie commented on code/datums/beam.dm:
```diff
@@ -89,28 +42,39 @@
 /datum/beam/Destroy()
 	Reset()
 	qdel(visuals)
+	UnregisterSignal(origin, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(target, COMSIG_MOVABLE_MOVED)
```
This should already happen at the base of `/datum/proc/Destroy()`

Couls committed:
Reviews, Turf layers, Projectiles and shove improvements

Couls committed:
Update gulag_item_reclaimer.dm

AutisticFroggy commented:
Keep up the good fight you madman.

Couls committed:
Jetpacks unregistering signals

Couls committed:
Update jetpack.dm

trexter555 commented:
its laggy as fuck

Jared-Fogle commented:
This was tested on high pop Bagil.
- Piggybacking doesn't work. I get the off hands, but don't climb and don't go with them. I can even drop my individual off hands.
- Even at both 60 and 144 (my refresh rate) FPS it's *unplayably* stuttery and laggy.
- Ghosts orbiting is very offset.

ShibaInuLord commented:
pain

ShizCalev commented:
seems really fucking laggy at 60fps, far beyond normally.

TheSilverNuke commented:
![2020-07-28 21_24_00-_tg_Station Basil  ENGLISH   US-WEST   100% FREE LAG _ PIXEL HELL STATION 13](https://user-images.githubusercontent.com/36066032/88755687-f05be600-d11e-11ea-952b-33c952b64613.png)

Due to how mech hitboxes work plus pixel movement you can't see half of it

ishitbyabullet commented:
So, how are simplemobs going to react to this?

Couls committed:
Testmerge fixes

Solars center when anchored, orbiting follows the target now. Following is smooth but the ghost animation itself is kind of jank, glide_size for smoothing at higher fps and mechs can't turn at lightspeed to spam noise

Couls commented:
> So, how are simplemobs going to react to this?

They use pixel movement too

Jared-Fogle commented on code/game/mecha/mecha.dm:
```diff
@@ -628,7 +629,8 @@
 	setDir(direction)
-	if(turnsound)
+	if(turnsound && turn_cooldown < world.time)
+		turn_cooldown = world.time + 0.25 SECONDS
```
I believe you should be using the cooldown defines here.

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
Smoother movement and smooth ghost orbiting

Couls committed:
Update cooldowns to modern code

Couls committed:
Update solar.dm

Couls committed:
pixel phasing

Couls committed:
Revert this commit if movement isn't smoother in the testmerge

Couls committed:
oops

Couls committed:
Smooth pulling

Couls committed:
Update atoms_movable.dm

nemvar commented on code/game/objects/effects/spiders.dm:
```diff
@@ -84,8 +84,7 @@
 /obj/structure/spider/eggcluster/Initialize()
-	pixel_x = rand(3,-3)
-	pixel_y = rand(3,-3)
+	forceStep(null, step_x, step_y)
```
```
	if(loc)
		forceStep(null, rand(3,-3), rand(3,-3))

nemvar commented on code/game/objects/items/toys.dm:
```diff
@@ -882,7 +882,7 @@
 	var/cardname = null
 	var/flipped = 0
-	pixel_x = -5
+	step_x = -5
```
These look like actual visual effects. pixel_x is probably fine here.

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
thanks nemvar!

Couls committed:
Vending machines tweak

Couls committed:
Testmerge #2 Fixes

Things fixed:
Spacemove spams the "you push off [x]" message
Ore box too big
tank dispenser too big
hitting directional windows
Clearing blockages when theres only 1 tile free
Throwing things hitting grille through window
Disposals outlet stopping garbage
Trying to orbit someone takes you to nullspace sometimes
Changes:
round step_size so people don't move at decimals of pixels, speed changes should be more significant
Lower the run speed to 6 px a step from 8 px a step

Couls committed:
Performance changes

TILE_BOUND is much cheaper than not having it, but causes  bugs with opaque objects so we're going to keep TILE_BOUND on everything but mobs and opaque movables

Limits the amount of atoms to check when we try to sidestep

Spellcheck one of the autodoc comments

Valtosin commented:
- Simple mobs are too fast, such as lavaland fauna: https://youtu.be/JbSp43oyawA?t=785

Couls commented:
Tested this out, looks like you were just stunned by the goliaths I was only hit once by a group of two https://streamable.com/xwio0r

Valtosin commented:
I meant they move faster than the original tile-based goliaths. But yeah, that is just balancing problem I think.

cacogen commented:
we live in the future

LemonInTheDark commented:
I want to believe

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

ayebee123 commented:
haven't played on the server in a few days, nor have i tested this myself, but from watching the goliath clip i think that this might make the crusher's backstab bonus somewhat redundant, since the goliaths don't seem to stop for anything and almost always face the player

this is probably just a dumb thing that'll be dealt with eventually, but it's just something thats on my mind

Couls committed:
fixies

Valtosin commented:
In addition to simple mobs: https://cdn.discordapp.com/attachments/723312598778380401/742431446714548324/Fix_this_SHIT.webm

Couls commented:
what the heck lmao

Couls committed:
Inertia stuff

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
new sidestep handling, a bunch of misc fixes

Couls committed:
fireman carry and piggyback fixes

toggle sidestep pref, remember to add it to save files before merge
prevent wasteful moves from AI(stop walking if you can't move)

Couls committed:
Remove dodging when approaching target

Just doesn't work with pixel movement and doesn't make much sense

Couls committed:
default edge sliding to on

Couls commented:
New edge sliding preference, makes it so you automatically slide along corners you walk into if more than half of your hitbox is past the edge

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
use bound_width and bound_height instead of bounds var

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
Speed up bot patrols, mulebot fixes, spraybottle fixes, edge slide pref transfers over better

Couls committed:
mechs can sidestep, a bit of doc for colliders, clean up old code

Couls committed:
Update window.dm

Couls committed:
Pixel movement document with some pointers and tips and tear in reality fix

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Valtosin commented:
It seems chemical pipes somehow doesn't connect to chemical machinery: 
![](https://i.imgur.com/0aVV1qt.png)
Sometimes pulled objects become invisible: https://lambda.sx/LJ3.mp4

Couls commented:
I think the pipes have to do with layering but i'll look into the pulled object thing

Couls committed:
fix morgue trays being JANK

Valtosin commented:
You can also seat/lie on something and if you on the edge of tile, you buckle to air. Result:
![image](https://user-images.githubusercontent.com/10297716/90197659-781d4200-ddf9-11ea-99b8-39c6c5225120.png)

trerri commented:
merge this with no changelog on april fools

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
Move shoving vars over to carbon fix errors

LemonInTheDark commented:
Oh dear god

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls commented:
> Oh dear god

god won't help you here

Couls committed:
Merge branch 'pixelmovementsubmission' into pixelmovement-inertiabranch

LemonInTheDark commented on code/modules/mob/living/carbon/carbon_defines.dm:
```diff
@@ -89,5 +89,11 @@
+
 	/// Can other carbons be shoved into this one to make it fall?
 	var/can_be_shoved_into = FALSE
+	/// Are they being shoved?
+	var/shoved = FALSE
```
this feels like it should be a bitflag

Couls replied:
what would the benefits be?

LemonInTheDark replied:
Performance concerns apparently

Couls committed:
fixes

LemonInTheDark commented:
I am happy to report that coul has not snuck poolcode into one of the last 11 commits, and nothing seems horribly out of order. Let's give this a testmerge once he fixes up a bounding box

Couls commented:
I'd like to raise the world fps in this PR but I want to be fully informed on whether or not this would be possible, the pros and cons and basically why it would be a bad idea, any input would be appreciated

currently the pros seen by a small test of raising the world fps to 40 was much smoother movement and less jitter in movement. I assume the server load will be higher but I was told all the subsystems would tick at the same rate. I think raising the world fps is an important step in making pixel movement smooth and solving the problem people are noticing with stutter and jitter

thank you for reading my 10th grade essay
yours truly, coul

JenqaDev commented:
merge when

Couls commented:
Getting the fps to 40 is probably unlikely, I'll see what I can do about pushing 25 fps and finding a different way to make movement feel less jank i'm still working on this but don't have the same kind of time since my school semester has started once again

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
remove smooth movement, it won't work with pixel move

Couls committed:
Merge branch 'pixelmovementsubmission' into pixelmovement-inertiabranch

Couls committed:
fix orbiting, inertia spacedrift, fix runtime

Couls committed:
Update orbiter.dm

Couls commented:
Pixel movement now uses an inertia system as of this comment, there are a few major changes:
SSmovement subsystem that handles inertia of all moving mobs and removes them from the processing when they stop moving
add_velocity helper which is under movables and is passed a direction and velocity and adds inertia in that direction
spacedrift and spacewind now use this system, pulling and mob movement use this system, client-less mobs and simple animals still use steps and walks

I'll add this to the readme next time I work on this

Acer202 commented:
I downloaded this code to try out the inertia thing. I'm a supporter of the pixel movement idea, but I'm not sure how I feel about movement inertia; it feels imprecise and admittedly pretty awkward. Was the purpose of it to smooth out movement?

Couls commented:
> I downloaded this code to try out the inertia thing. I'm a supporter of the pixel movement idea, but I'm not sure how I feel about movement inertia; it feels imprecise and admittedly pretty awkward. Was the purpose of it to smooth out movement?

Yes, the idea is to have acceleration and deceleration to reduce jitter and make movement feel more consistent and smooth, there are ways to improve it and its highly configurable, if you find its too slidey, simply raise the decel var on the mob you're controlling, if you find its too slow to speed up, raise the acceleration and if it feels very slow then you can modify the walk and run delay in the config, during a private test I got a lot of feedback and a few people mentioned the same thing but were satisfied with a few modifications(3 run delay and 2 decel 1 accel)

Acer202 commented:
Thanks, I'll give it a shot!

Couls committed:
lower the delays for walking and running slightly

LordSaladin commented:
testmerge on golden.

Couls committed:
default sidestepping to off, raise fps by 5

stylemistake commented:
Coul, why you do this to me...

Couls committed:
revert tgui changes, experimental pulling stuff

Couls commented:
Ahhh problem of the week: Figuring out a way to handle objects getting stuck in corners gracefully. If someone has any ideas or suggestions for this that would be very welcome.
There is edge sliding code in place, but the problem is that the code itself isn't really meant for pullable objects and the window where the edge would be corrected is very small(and with good reason because if you slide a large distance  it looks very jumpy and jittery when this happens multiple times)

If some brave soul wants to help me with this
The relevant edge sliding code file: https://github.com/tgstation/tgstation/blob/0ab13d45beb3d3c30eed1abd959698b6ba615f42/code/game/collider.dm
Pulling code:
https://github.com/tgstation/tgstation/blob/0ab13d45beb3d3c30eed1abd959698b6ba615f42/code/game/atoms_movable.dm#L347

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
fix edge sliding on pulled objects

Couls committed:
fix conveyors registering colliders as movers

Couls committed:
heater bounds, add velocity to the pixelmovement readme

Couls committed:
Fix vehicles potentially crashing the server, edge sliding is on by default again

Couls committed:
Revert FPS stuff, mech fixes, bot fixes, pull improvements, SSmovement improvements

Canisters also snap when wrenched to connectors

Couls commented:
Should be good for a testmerge. Need to get feedback and see how this implementation handles on higher pops

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

MrMelbert commented:
Okay, so, a few notes from the manuel TM:
1. Pushing and pulling is such a crapshoot. Hydroponics trays cannot be pushed, so you can GBJ yourself really easily. Things just warp around and it's not fun at all.
2. You can move while climbing onto tables, and you can slip / get caltrops by things on tables. if they're on the side of it.
4. Slippery things from botany and punji sticks have massive hitboxes
4. The center of the screen has changed and overlays like the blind overlay weren't adjusted to this
I recorded most of my round, and i'll edit this comment with a link to the footage when it's rendered

Couls committed:
only update parallax on turf changes

Couls committed:
Temporarily disable turf decals, fix machinery being unpullable

Couls committed:
raise deceleration to 2 to that people feel less slidey especially at higher ping

Couls committed:
speed up ghosts

Couls committed:
more movement means more map updates which means we need to reserve more of the tick for sendmaps

Couls committed:
Update _tick.dm

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

SpaceManiac commented:
This PR is enormous, impossible to review, nearly 2 months old, certainly buggy by its size alone, and controversial.

:+1: 29, 3 maintainers
- LemonInTheDark
- actioninja
- stylemistake 

:-1: 38, 4 maintainers
- ShizCalev
- Jared-Fogle
- stylemistake (@stylemistake so do you want it or not?)
- SpaceManiac

I don't want to make light of the extreme amount of effort that has gone into this PR, but I don't want this feature and I'm not convinced our players will want this feature or the inevitable period of overall reduced game-playability, bugginess, and balance issues that will come with its merge.

Couls commented:
Not sure what I can say or do at the moment to convince you, I do plan to continue working on it so make that period of buginess much shorter than possible and I can say during the testmerges more and more people have come around and seen the potential of something like this, it’s hard to say that it’s worth it now when I can’t show you the future or what this will allow.

There will definitely be balance features but I think pixel movement will absolutely allow for a more balanced game in the long run as I already have lots of ideas how to deal with a few of what may be the more annoying changes(some that wouldn’t be hard at all to implement) for examples a swing animation that creates a small arc in a 180 degree angle a FEW pixels in front of the attacker to remove pixel hunting but not create essentially a circle of death. Projectile hitboxes can be adjusted to be as big as we want so if players are getting frustrated with missing their opponents there can be one tile room for error. Updating bounds for objects is something trivial and I’ll gladly do alone as it doesn’t require much work and you can dig through the dmi editor to get the exact pixel coords of every object and then manually input it

My point is I wouldn’t want this to go through in an unplayable state which is why it’s been around 2 months and while the transitional phase may be a bit disorienting I do think it’s better for the health of the game and allows for a lot of stuff that was previously impossible that I think a lot of players will enjoy.

That said I completely understand if this is still unwanted and closed but I do care about this one a lot and wish that everyone(including maintainers) would keep an open mind and try to see this through with me. So that it can at least get a fair shot, it’s gone through a few iterations and I think it’s close to being a smooth experience. The two months of dedicated development, with testmerges and local tests should prove that I’m interested in maintaining this

stylemistake commented:
I put a like because pixel movement lools like the right way to go, especially with the addition of newtonian physics, it can result in a lot of cool interactions in the future. But I put dislike mostly because of the neverending bugs that are bound to happen, and they just have to be addressed. I hope Coul figures this out.

Couls committed:
Testmerge fixes 3

Hunger is now based off how many pixels you move, lithium now moves you more significantly, vehicles now have slightly different handling for that "drifty" feel without making humans feel like they're on an ice level, skateboard ollie behaviour modified(now ollies a full turf from where you start) grinding with skateboard uses inertia now

Couls committed:
all grown plants are now 8 pixels by 8 pixels until I have time to properly go through and do each plant, runechat is now properly centered

wesoda25 commented:
> I don't want to make light of the extreme amount of effort that has gone into this PR, but I don't want this feature and I'm not convinced our players will want this feature or the inevitable period of overall reduced game-playability, bugginess, and balance issues that will come with its merge.

You’re going against the status quo dude.

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

Couls committed:
remove SSconveyors and just use fastprocess like master

Couls commented:
Still being worked on, kinda a busy week but I'll resolve conflicts this weekend

Couls committed:
Merge remote-tracking branch 'upstream/master' into pixelmovementsubmission

actioninja commented:
coul mentioned on the discord they're going to be too busy to work on this so I'm closing this for now.
It can be reopened when more ready or a new pr can be opened if someone else wants to take the reigns.

Jared-Fogle commented:
rip

ShibaInuLord commented:
Thank heck

Valtosin commented:
Yes, let's continue

stylemistake commented:
Necromancers.

SweptWasTaken commented:
Wow this is just like Space Station 14!

LemonInTheDark commented:
stfus

cowbot92 commented:
2025 bringing this back 👀 ?

L0pz commented:
REVIVE IT
